Mathlib/Analysis/InnerProductSpace/Positive.lean

```diff
@@ -206,6 +206,98 @@ end PartialOrder
 
 end ContinuousLinearMap
 
+/-- a star projection is an idempotent and self-adjoint element -/
+structure IsStarProjection {M : Type*} [Mul M] [Star M] (p : M) : Prop where
+  isIdempotentElem : IsIdempotentElem p
+  isSelfAdjoint : IsSelfAdjoint p
+
+/-- a star projection is non-negative in a star-ordered ring -/
+theorem IsStarProjection.nonneg {M : Type*} [NonUnitalSemiring M] [PartialOrder M]
+  [StarRing M] [StarOrderedRing M] {p : M}
+  (hp : IsStarProjection p) : 0 ≤ p :=
+by
+  rw [← hp.isIdempotentElem]
+  nth_rw 1 [← hp.isSelfAdjoint]
+  exact star_mul_self_nonneg p
+
+-- this will be removed in the future when we have `StarOrderedRing (E →L[𝕜] E)`
+-- as this is just `IsStarProjection.nonneg`
+/-- a star projection operator is positive -/
+theorem isStarProjection_isPositive [CompleteSpace E] {p : E →L[𝕜] E}
+  (hp : IsStarProjection p) : 0 ≤ p :=
```

Comment:
Maybe the comment should be:
1. located in the docstring so it doesn't get lost or forgotten
2. instead say not that it will be removed, but that the proof can be simplified when the `StarOrderedRing` instance exists

and the conclusion should be `p.IsPositive`.